src/Grpc.Core.Api/Status.cs

```diff
@@ -0,0 +1,95 @@
+#region Copyright notice and license
+// Copyright 2015 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using System;
+
+namespace Grpc.Core
+{
+    /// <summary>
+    /// Represents RPC result, which consists of <see cref="StatusCode"/> and an optional detail string.
+    /// </summary>
+    public struct Status
```

Comment:
Should make this readonly

Reply:
added a TODO since I only want to make optional changes after everything is setup and works well.